components/blobserve/cmd/run.go

```diff
@@ -100,6 +105,31 @@ var runCmd = &cobra.Command{
 			log.WithField("addr", cfg.PrometheusAddr).Info("started Prometheus metrics server")
 		}
 
+		if cfg.ReadinessProbeAddr != "" {
+			// use the first layer as source for the tests
+			if len(cfg.BlobServe.Repos) < 1 {
+				log.Panic("the configuration file do not contain static layers")
+			}
+
+			var repository string
+			for k := range cfg.BlobServe.Repos {
+				repository = k
+			}
+			staticLayerHost := strings.Split(repository, "/")[0]
+			// Ensure we can resolve DNS queries, can access gcr.io and the local Registry is up and running
+			health := healthcheck.NewHandler()
+			health.AddReadinessCheck("dns", kubernetes.DNSCanResolveProbe(staticLayerHost, 1*time.Second))
+			health.AddReadinessCheck("registry", kubernetes.NetworkIsReachableProbe(fmt.Sprintf("http://%v", repository)))
+			health.AddLivenessCheck("dns", kubernetes.DNSCanResolveProbe(staticLayerHost, 1*time.Second))
+			health.AddLivenessCheck("registry", kubernetes.NetworkIsReachableProbe(fmt.Sprintf("http://%v", repository)))
```

Comment:
Do we really want blobserve to go down if we can no longer reach those? Chances are things are cached and blobserve would just keep functioning.

Reply:
Sure, we can remove those. Now, How do we know there's an error in blobserve without a restart?

Reply:
We would want to add metrics/alerts for the error cases we know.